haystack/question_generator/question_generator.py

```diff
@@ -0,0 +1,98 @@
+from transformers import AutoModelForSeq2SeqLM
+from transformers import AutoTokenizer
+from haystack import BaseComponent
+from haystack.preprocessor import PreProcessor
+
+
+class QuestionGenerator(BaseComponent):
+    """ The Question Generator takes only a document as input and outputs questions that it thinks can be
+    answered by this document. In our current implementation, input texts are split into chunks of 50 words
+    with a 10 word overlap. This is because the default model `valhalla/t5-base-e2e-qg` seems to generate only
+    about 3 questions per passage regardless of length. Our approach prioritizes the creation of more questions
+    over processing efficiency (T5 is able to digest much more than 50 words at once). The returned questions
+    generally come in an order dictated by the order of their answers i.e. early questions in the list generally
+    come from earlier in the document."""
+
+    outgoing_edges = 1
+
+    def __init__(self,
+                 model_name_or_path="valhalla/t5-base-e2e-qg",
```

Comment:
it would be nice to suggest user what type of model are supported like T5, GPT or we can tell text generation models only used here.

Reply:
Agreed. Maybe this link could help here as a comment: https://huggingface.co/models?filter=question-generation